src/uct/base/uct_md.h

```diff
@@ -274,9 +260,20 @@ typedef struct uct_md_mem_alloc_params {
     unsigned          flags;
 } uct_md_mem_alloc_params_t;
 
-typedef uct_md_mem_alloc_params_t* uct_md_mem_alloc_params_h;
-
-ucs_status_t uct_md_mem_alloc_v2(uct_md_h md, uct_md_mem_alloc_params_h params);
+/**
+ * @ingroup UCT_MD
+ * @brief Allocate memory for zero-copy sends and remote access.
+ *
+ * Allocate memory on the memory domain. In order to use this function, MD
+ * must support @ref UCT_MD_FLAG_ALLOC flag.
+ *
+ * @param [in]     md          Memory domain to allocate memory on.
+ * @param [in,out] params      Struct containing parameters such as memory type, address, allocation length, 
+ *                             and other parameters from @ref uct_md_mem_alloc function.
+ * @param [out]    memh_p      Filled with handle for allocated region.
+ */
+ucs_status_t uct_md_mem_alloc_v2(uct_md_h md, uct_md_mem_alloc_params_t *params,
```

Comment:
Actually this header is not an "exposed" API at all. Maybe we actually need `v2` for `uct_mem_alloc` - which is an API func?

Reply:
Or maybe we can just change `uct_mem_alloc ` definition from:
```c
 status = uct_md_mem_alloc(md, &alloc_length, &address,
                                          mem_type, flags, alloc_name,
                                          &memh);
```
To:
```c
  uct_md_mem_alloc_params_t params = {
     .length = &alloc_length,
     .flags   = flags,
     ...
  }
 status = uct_md_mem_alloc_v2(md, params, &memh);
```